docs/guides/03-web-components.md

```diff
@@ -0,0 +1,94 @@
+## Web components
+
+Airship provides a set of [web components](https://www.webcomponents.org/introduction). One of the key features of the Web Components standard is the ability to offer custom elements that encapsulate functionality on an HTML page, rather than having to make do with a long, nested batch of elements that together provide a custom page feature.
+
+This is especially interesting to create geospatial widgets, such as a histogram or a category-view where instead of writing all the code by hand every time you can use just a simple a custom element.
+
+
+```html
+<as-category-widget
+  heading="Business Volume"
+  description="Description"
+  default-bar-color="#47DB99">
+</as-category-widget>
+```
+
+### Attributes
+
+[Attributes](https://en.wikipedia.org/wiki/HTML_attribute) modify the default functionality of an element. Usually the attributes will have a default value and using them will be optional. In the example above, heading and description are required attributes while the bar color is optional. 
+
+HTML Attributes can be only of type `string`, so if you want to use an object as an attribute you need to use javascript.
+
+Although you can use the [setAttribute](https://developer.mozilla.org/en-US/docs/Web/API/Element/setAttribute) method, you can directly assign the attributes as properties of the element.
+
+```js
+const $categoryWidget = document.querySelector('as-category-widget');
+const categoriesList = [
+  { name: 'A New Hope', value: 359029623, color: '#FABADA' },
+  { name: 'The Empire Strikes Back', value: 236513856 },
+  { name: 'Return of the Jedi', value: 204338075 },
+];
+
+// Same as: $categoryWidget.setAttribute('categories', categoriesList);
+$categoryWidget.categories = categoriesList;
+```
+
+### Loading web components
+
+Loading and using web components is very easy, you just need to include our component loader in the head of your app. Once the loader was included It will automatically detect the use of web-components and load them on demand. It also detects browser capabilities and will automatically decide whether or not to include polyfills, leading to smaller bundles in modern browsers.
```

Comment:
Once the loader is included, it will automatically detect the use of web-components and load them on demand.